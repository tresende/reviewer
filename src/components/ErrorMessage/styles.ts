import styled, { css } from 'styled-components'
import media from 'styled-media-query'

export const Wrapper = styled.div`
  ${media.lessThan('small')`
    display: flex;
    flex-direction: column;
    text-align: center;
  `}
`

export const ErrorMessage = styled.span`
  ${({ theme }) => css`
    position: absolute;
    max-width: 24rem;
    top: 5rem;
    left: 4rem;
    font-size: ${theme.font.sizes.xxlarge};
    font-weight: bold;
    ${media.lessThan('small')`
      max-width: 100%;
      position: initial;
    `}
  `}
`

export const Image = styled.img`
  background-image: url('/img/error.png');
  background-position: center;
  width: 100%;
  height: 100%;
  background-repeat: no-repeat;
  background-size: 100%;
`
