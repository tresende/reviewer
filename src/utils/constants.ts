const SCORE_LABEL = ['PÃ©ssimo', 'Ruim', 'Mediano', 'Bom', 'Perfeito']
const MIN_TEXT_LENGTH = 5
const MAX_RATING = 5
const DEFAULT_SCORE = 3
const ANIMATION_DURATION = 1000

export { SCORE_LABEL, MAX_RATING, ANIMATION_DURATION, MIN_TEXT_LENGTH, DEFAULT_SCORE }
