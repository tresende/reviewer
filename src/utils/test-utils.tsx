import { ReactElement } from 'react'
import { render, RenderOptions } from '@testing-library/react'

import { ThemeProvider } from 'styled-components'
import theme from 'styles/theme'
import { LoadingContext, LoadingContextData, LoadingContextDefaultValues } from 'hooks/use-loading'

type CustomRenderProps = {
  loadingProviderProps?: LoadingContextData
} & Omit<RenderOptions, 'queries'>

const customRender = (
  ui: ReactElement,
  { loadingProviderProps = LoadingContextDefaultValues, ...renderOptions }: CustomRenderProps = {}
) =>
  render(
    <ThemeProvider theme={theme}>
      <LoadingContext.Provider value={loadingProviderProps}>{ui}</LoadingContext.Provider>
    </ThemeProvider>,
    renderOptions
  )

export * from '@testing-library/react'
export { customRender as render }
