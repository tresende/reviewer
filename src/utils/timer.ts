export function timeout(ms?: number | undefined) {
  return new Promise((resolve) => setTimeout(resolve, ms))
}
