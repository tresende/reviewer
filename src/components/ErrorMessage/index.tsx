import * as S from './styles'

const ErrorMessage = () => (
  <S.Wrapper>
    <S.ErrorMessage>
      Ocorreu um erro!
      <br />
      Tente novamente mais tarde
    </S.ErrorMessage>
    <S.Image src="/img/error.png"></S.Image>
  </S.Wrapper>
)

export default ErrorMessage
