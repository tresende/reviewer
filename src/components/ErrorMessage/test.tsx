import { render } from 'utils/test-utils'
import ErrorMessage from '.'

describe('<ErrorMessage />', () => {
  it('should render component correctly', () => {
    const { container } = render(<ErrorMessage />)
    expect(container.firstChild).toMatchSnapshot()
  })
})
