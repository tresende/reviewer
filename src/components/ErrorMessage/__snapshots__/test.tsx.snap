// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ErrorMessage /> should render component correctly 1`] = `
.c1 {
  position: absolute;
  max-width: 24rem;
  top: 5rem;
  left: 4rem;
  font-size: 2.8rem;
  font-weight: bold;
}

.c2 {
  background-image: url('/img/error.png');
  background-position: center;
  width: 100%;
  height: 100%;
  background-repeat: no-repeat;
  background-size: 100%;
}

@media (max-width:450px) {
  .c0 {
    display: -webkit-box;
    display: -webkit-flex;
    display: -ms-flexbox;
    display: flex;
    -webkit-flex-direction: column;
    -ms-flex-direction: column;
    flex-direction: column;
    text-align: center;
  }
}

@media (max-width:450px) {
  .c1 {
    max-width: 100%;
    position: initial;
  }
}

<div
  class="c0"
>
  <span
    class="c1"
  >
    Ocorreu um erro!
    <br />
    Tente novamente mais tarde
  </span>
  <img
    class="c2"
    src="/img/error.png"
  />
</div>
`;
